contrib/AutoGrouper.lua

```diff
@@ -0,0 +1,155 @@
+ 2) Require this file in your luarc file, as with any other dt plug-in
+
+Set a gap amount in second which will be used to determine when images should no 
+longer be added to a group. If an image is more then the specified amount of time
+from the last image in the group it will not be added. Images without timestamps 
+in exif data will be ignored.
+
+There are two buttons. One allows the grouping to be performed only on the currently
+selected images, the other button performs grouping on the entire active collection
+]]
+
+local dt = require "darktable"
+local mod = 'autogroup'
+local gettext = dt.gettext
+-- Tell gettext where to find the .mo file translating messages for a particular domain
+gettext.bindtextdomain("AutoGrouper",dt.configuration.config_dir.."/lua/locale/")
+local function _(msgid)
+    return gettext.dgettext("AutoGrouper", msgid)
+end
+
+local function InRange(test, low, high) --tests if test value is within range of low and high (inclusive)
+	if test >= low and test <= high then
+		return true
+	else
+		return false
+	end
+end
+
+local function CompTime(first, second) --compares the timestamps and returns true if first was taken before second
+	first_time = first.exif_datetime_taken
+	if string.match(first_time, '[0-9]') == nil then first_time = '9999:99:99 99:99:99' end
+	first_time = tonumber(string.gsub(first_time, '[^0-9]*',''))
+	second_time = second.exif_datetime_taken
+	if string.match(second_time, '[0-9]') == nil then second_time = '9999:99:99 99:99:99' end
+	second_time = tonumber(string.gsub(second_time, '[^0-9]*',''))
+	return first_time < second_time
+end
+
+local function SeperateTime(str) --seperates the timestamp into individual components for used with OS.time operations
+	local cleaned = string.gsub(str, '[^%d]',':')
+	cleaned = string.gsub(cleaned, '::*',':')  --YYYY:MM:DD:hh:mm:ss
+	local year = string.sub(cleaned,1,4)
+	local month = string.sub(cleaned,6,7)
+	local day = string.sub(cleaned,9,10)
+	local hour = string.sub(cleaned,12,13)
+	local min = string.sub(cleaned,15,16)
+	local sec = string.sub(cleaned,18,19)
+	return {year = year, month = month, day = day, hour = hour, min = min, sec = sec}
+end
+
+local function GetTimeDiff(curr_image, prev_image) --returns the time difference (in sec.) from current image and the previous image
+	local curr_time = SeperateTime(curr_image.exif_datetime_taken)
+	local prev_time = SeperateTime(prev_image.exif_datetime_taken)
+	return os.time(curr_time)-os.time(prev_time)
+end
+
+local function main(on_collection)
+	local images = {}
+	if on_collection then 
+		local col_images = dt.collection
+		for i,image in ipairs(col_images) do --copy images to a standard table, table.sort barfs on type dt_lua_singleton_image_collection
+			table.insert(images,i,image)
+		end
+	else
+		images = dt.gui.selection()
+	end
+	dt.preferences.write(mod, 'active_gap', 'integer', GUI.gap.value)
+	if #images < 2 then return end

```

Comment:
Maybe print a message to the screen (and/or log) about insufficient images instead of just silently doing nothing